tests/test_provider.py

```diff
@@ -14,3 +15,65 @@ def test_report_info_od(provider):
     pinfo2 = provider.get_quota()
     assert pinfo2['used'] > 0
     assert pinfo2['limit'] > 0
+
+def test_interrupted_file_upload(provider):
+    # file_upload_size = 4 MiB, should take 12/4=3 API calls to upload file
+    file_size = 12 * 1024 * 1024
+    data = BytesIO(os.urandom(file_size))
+    dest = provider.temp_name("dest")
+
+    def hit_api(action, path=None, url=None, data=None, headers=None, json=None):
```

Comment:
Let's have a conversation about "hit_api" to see if there is a simpler way to express this... I have some thoughts